ansible_collections/arista/avd/plugins/filter/range_expand.py

```diff
@@ -10,128 +10,69 @@
 from ansible.errors import AnsibleFilterError
 
 
-def range_expand(range_to_expand):
-    if not (isinstance(range_to_expand, list) or isinstance(range_to_expand, str)):
-        raise AnsibleFilterError(f"value must be of type list or str, got {type(range_to_expand)}")
+def expand_parentheses(range_sr, one_range):
+    groups = range_sr.groups()
+    start_index, end_index = range_sr.start(), range_sr.end()
+    replaced_list = []
+    if groups[0]:
+        # To replace the values inside parenthesis, Ex. (2,3,4-6)
+        ports_list = groups[0].replace("(", "").replace(")", "").split(",")
+        for i in ports_list:
+            replaced_list.append(one_range.replace(one_range[start_index:end_index], str(i), 1))
+    return replaced_list
+
+
+def expand_hypen(range_sr, one_range):
+    groups = range_sr.groups()
+    start_index, end_index = range_sr.start(), range_sr.end()
+    replaced_list = []
+    if groups[0]:
+        # To replace the values of range, Ex. 4-6
+        range_start, range_end = groups[0].split("-")
+        for i in range(int(range_start), int(range_end) + 1):
+            replaced_list.append(one_range.replace(one_range[start_index:end_index], str(i), 1))
+    return replaced_list
+
+
+def range_expand(one_range, prefix=""):
+    if not (isinstance(one_range, list) or isinstance(one_range, str)):
+        raise AnsibleFilterError(f"value must be of type list or str, got {type(one_range)}")
 
     result = []
 
     # If we got a list, unpack it and run this function recursively
-    if isinstance(range_to_expand, list):
-        for r in range_to_expand:
-            result.extend(range_expand(r))
+    if isinstance(one_range, list):
+        for r in one_range:
```

Comment:
please avoid abbreviated vars, since it makes it harder to read the code. But also avoid reserved words when you use full words :)

Reply:
Fixed